grype/pkg/package_test.go

```diff
@@ -0,0 +1,229 @@
+				ManifestName:  "main-section-name-info",
+			},
+		},
+		// below cases we assert that there is no mapped metadata
+		{
+			name: "apk-metadata",
+			syftPkg: syftPkg.Package{
+				MetadataType: syftPkg.ApkMetadataType,
+				Metadata: syftPkg.ApkMetadata{
+					Package:          "a",
+					OriginPackage:    "a",
+					Maintainer:       "a",
+					Version:          "a",
+					License:          "a",
+					Architecture:     "a",
+					URL:              "a",
+					Description:      "a",
+					Size:             1,
+					InstalledSize:    1,
+					PullDependencies: "a",
+					PullChecksum:     "a",
+					GitCommitOfAport: "a",
+				},
+			},
+			metadata: nil,
+		},
+		{
+			name: "npm-metadata",
+			syftPkg: syftPkg.Package{
+				MetadataType: syftPkg.NpmPackageJSONMetadataType,
+				Metadata: syftPkg.NpmPackageJSONMetadata{
+					Author:      "a",
+					Homepage:    "a",
+					Description: "a",
+					URL:         "a",
+				},
+			},
+			metadata: nil,
+		},
+		{
+			name: "python-metadata",
+			syftPkg: syftPkg.Package{
+				MetadataType: syftPkg.PythonPackageMetadataType,
+				Metadata: syftPkg.PythonPackageMetadata{
+					Name:                 "a",
+					Version:              "a",
+					License:              "a",
+					Author:               "a",
+					AuthorEmail:          "a",
+					Platform:             "a",
+					SitePackagesRootPath: "a",
+				},
+			},
+			metadata: nil,
+		},
+		{
+			name: "gem-metadata",
+			syftPkg: syftPkg.Package{
+				MetadataType: syftPkg.GemMetadataType,
+				Metadata: syftPkg.GemMetadata{
+					Name:     "a",
+					Version:  "a",
+					Homepage: "a",
+				},
+			},
+			metadata: nil,
+		},
+		{
+			name: "kb-metadata",
+			syftPkg: syftPkg.Package{
+				MetadataType: syftPkg.KbPackageMetadataType,
+				Metadata: syftPkg.KbPackageMetadata{
+					ProductID: "a",
+					Kb:        "a",
+				},
+			},
+			metadata: nil,
+		},
+		{
+			name: "rust-metadata",
+			syftPkg: syftPkg.Package{
+				MetadataType: syftPkg.RustCargoPackageMetadataType,
+				Metadata: syftPkg.CargoPackageMetadata{
+					Name:     "a",
+					Version:  "a",
+					Source:   "a",
+					Checksum: "a",
+				},
+			},
+			metadata: nil,
+		},
+	}
+
+	// capture each observed metadata type, we should see all of them relate to what syft provides by the end of testing
+	observedMetadataTypes := set.NewStringSet()
+	expectedMetadataTypes := set.NewStringSet()
+	for _, ty := range syftPkg.AllMetadataTypes {
+		expectedMetadataTypes.Add(string(ty))
+	}
+
+	// run all of our cases
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			observedMetadataTypes.Add(string(test.syftPkg.MetadataType))
+			assert.Equal(t, test.metadata, New(&test.syftPkg).Metadata)
+		})
+	}
+
+	// did we see all possible metadata types? if not, then there is an uncovered case and this test should error out
+	if !expectedMetadataTypes.IsEqual(observedMetadataTypes) {
+		t.Errorf("did not observe all possible package metadata types: missing: %+v extra %+v",
```

Comment:
nit: Given the colon after "missing", I'm wondering if there should be a colon after "extra"